`cedar-policy-core` fails to compile without default features enabled

### Before opening, please confirm:


- [x] I have [searched for duplicate or closed issues](https://github.com/cedar-policy/cedar/issues?q=is%3Aissue+).
- [x] I have read the guide for [submitting bug reports](https://github.com/cedar-policy/cedar/blob/main/CONTRIBUTING.md#bug-reports).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### Bug Category

Other

### Describe the bug

When adding `cedar-policy-core` as dependency with `default-features = false` it fails to compile as in the `ast::name` module there is a `use regex::Regex` statement to validate the `Name`. However, `regex` is an optional dependency which is only enabled with the `datetime` or `decimal` feature.

### Expected behavior

It should compile without the default features.

### Reproduction steps

Compile the crate without default features

### Code Snippet

It's possible to fix this issue by making `regex` non-optional:
```diff
-regex = { version = "1.8", features = ["unicode"], optional = true }
+regex = { version = "1.8", features = ["unicode"] }
```


### Log output

```
error[E0432]: unresolved import `regex`
  --> /Users/_/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/cedar-policy-core-4.5.0/src/ast/name.rs:22:5
   |
22 | use regex::Regex;
   |     ^^^^^ use of unresolved module or unlinked crate `regex`
   |
   = help: if you wanted to use a crate named `regex`, use `cargo add regex` to add it to your `Cargo.toml`
```


### Additional configuration

_No response_

### Operating  System

_No response_

### Additional information and screenshots

_No response_

--- a/cedar-policy-core/Cargo.toml
+++ b/cedar-policy-core/Cargo.toml
@@ -30,9 +30,7 @@ miette = { version = "7.6.0", features = ["serde"] }
 nonempty = { version = "0.10.0", features = ["serialize"] }
 educe = "0.6.0"
 unicode-security = "0.1.0"
-
-# decimal extension requires regex
-regex = { version = "1.8", features = ["unicode"], optional = true }
+regex = { version = "1.8", features = ["unicode"]}
 
 # wasm dependencies
 serde-wasm-bindgen = { version = "0.6", optional = true }
@@ -46,8 +44,8 @@ chrono = { version = "0.4.40", optional = true, default-features = false}
 # by default, enable all Cedar extensions
 default = ["ipaddr", "decimal", "datetime"]
 ipaddr = []
-decimal = ["dep:regex"]
-datetime = ["dep:chrono", "dep:regex"]
+decimal = []
+datetime = ["dep:chrono"]
 
 # Enables `Arbitrary` implementations for several types in this crate
 arbitrary = ["dep:arbitrary"]

